src/EnergyPlus/DataPlant.hh

```diff
@@ -597,6 +601,7 @@ namespace DataPlant {
 		int IndexInLoopSidePumps; // If I'm a pump, this tells my index in PL(:)%LS(:)%Pumps
 		Real64 TempDesCondIn;
 		Real64 TempDesEvapOut;
+		std::shared_ptr< PlantComponent > compPtr;
```

Comment:
@EnergyArchmage Thanks for this information.  I am always happy to see your successes and issues you have come across with your refactors.

I have a couple thoughts about your approach:

 - I am hoping we move away "pretty much" fully from index-based operation, and instead move toward using all reference based (in my case a pointer) operations.  This will help us move away from things hanging globally altogether.
 - In your approach, I'm not sure the value added in using unique pointers over storing just the instances themselves.  We can do object-oriented operations/structures with an array of instances, we could just be limited if we wanted to do polymorphism for different fan types, for example.

I think given how quick I am hoping to move while in Chicago, I am inclined to put this in the topic bucket for post-Chicago.  I think the solution that @mbadams5 described if sufficiently safe and relatively easy to understand without a bunch of homework, and also I don't feel like it paints us into a bad corner, we can use it as a stepping stone to get to something more formal.  And that "more formal" thing is likely something related to smarter pointers as you have shown here or in my initial efforts, but we'll see.